internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -126,6 +137,136 @@ func (r *ClusterDeploymentsReconciler) Reconcile(req ctrl.Request) (ctrl.Result,
 	return r.updateState(ctx, clusterDeployment, cluster, nil)
 }
 
+func (r *ClusterDeploymentsReconciler) updateClusterMetadata(ctx context.Context, cluster *hivev1.ClusterDeployment, c *common.Cluster) error {
+
+	s, err := r.getAdminPasswordSecret(ctx, cluster, c)
+	if err != nil {
+		return err
+	}
+	k, err := r.getKubeConfigSecret(ctx, cluster, c)
+	if err != nil {
+		return err
+	}
+	cluster.Spec.Installed = true
+	cluster.Spec.ClusterMetadata = &hivev1.ClusterMetadata{
+		ClusterID: c.OpenshiftClusterID.String(),
+		InfraID:   c.ID.String(),
+		AdminPasswordSecretRef: corev1.LocalObjectReference{
+			Name: s.Name,
+		},
+		AdminKubeconfigSecretRef: corev1.LocalObjectReference{
+			Name: k.Name,
+		},
+	}
+	return r.Update(ctx, cluster)
+}
+
+func (r *ClusterDeploymentsReconciler) getAdminPasswordSecret(ctx context.Context, cluster *hivev1.ClusterDeployment, c *common.Cluster) (*corev1.Secret, error) {
+	s := &corev1.Secret{}
+	name := fmt.Sprintf(adminPasswordSecretStringTemplate, cluster.Name)
```

Comment:
Shouldn't you check if a name already exist in the spec? if the spec will have a different name, for example user created installed cluster.

Reply:
For our case, day 1 the spec won't be populated.
For day 2, it will be handled in a different patch